SD-K9/src/app/providers/svg-manager.service.spec.ts

```diff
@@ -1,26 +1,211 @@
+        spyOn(service, 'drawSVGPath').and.callThrough();
     });
 
     afterEach(() => {
         service = null;
+        httpMock.verify();
     });
 
     it('should be created', () => {
         expect(service).toBeTruthy();
     });
-});
\ No newline at end of file
+
+    it('#getSVG should get SVG from assets', () => {
+
+        service.getSVG('any/0').subscribe(floorplan => {
+            expect(floorplan).toEqual(dummySVG);
+        });
+
+        const request = httpMock.expectOne('../../assets/any/0.svg');
+        expect(request.request.method).toBe('GET');
+
+        request.flush(dummySVG);
+    });
+
+    it('#getWalkableNodes should return a list of walkable nodes from the svg', () => {
+        const building = 'any';
+        const floor = 0;
+
+        const expectedNodes = [
+            { id: '', building, floor, x: 8, y: 2 },
+            { id: '', building, floor, x: 2, y: 8 }
+        ];
+
+        service.getWalkableNodes(building, floor)
+            .then(nodes => {
+                expect(nodes.length).toEqual(2);
+                expect(nodes).toEqual(expectedNodes);
+            })
+            .catch((error) => {
+                fail(error);
+            });
+
+        const request = httpMock.expectOne('../../assets/any/0.svg');
+        expect(request.request.method).toBe('GET');
+
+        request.flush(dummySVG);
+    });
+
+    it('#getClassroom should return a classroom given an id', () => {
+        const classID = 'A';
+        const building = 'any';
+        const floor = 0;
+
+        const expectedClassroom = {
+            id: 'A',
+            building,
+            floor,
+            x: 2,
+            y: 2
+        }
+
+        service.getClassroom(classID, building, floor)
+            .then(classroom => {
+                expect(classroom).toEqual(expectedClassroom);
+            })
+            .catch((error) => {
+                fail(error);
+            });
+
+        const request = httpMock.expectOne('../../assets/any/0.svg');
+        expect(request.request.method).toBe('GET');
+
+        request.flush(dummySVG);
+    });
+
+    it('#getClassroom should return a classroom given an id', () => {
+        const classID = 'A';
+        const building = 'any';
+        const floor = 0;
+
+        const expectedClassroom = {
+            id: 'A',
+            building,
+            floor,
+            x: 2,
+            y: 2
+        }
+
+        service.getClassroom(classID, building, floor)
+            .then(classroom => {
+                expect(classroom).toEqual(expectedClassroom);
+            })
+            .catch((error) => {
+                fail(error);
+            });
+
+        const request = httpMock.expectOne('../../assets/any/0.svg');
+        expect(request.request.method).toBe('GET');
+
+        request.flush(dummySVG);
+    });
+
+    it('#getVerticalTransportation should return a vertical transportation given an id', () => {
```

Comment:
getVerticalTransportation requires mode, building and floor as well.
```suggestion
    it('#getVerticalTransportation should return a vertical transportation given an id, mode, building and floor', () => {
```